yt_dlp/extractor/craftsy.py

```diff
@@ -0,0 +1,73 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import json
+
+from .common import InfoExtractor
+from ..utils import (
+    get_element_by_id,
+    js_to_json,
+    traverse_obj,
+)
+
+
+class CraftsyIE(InfoExtractor):
+    _VALID_URL = r'https?://www.craftsy.com/class/(?P<id>[a-z0-9_-]+)/'
+    _TESTS = [{
+        'url': 'https://www.craftsy.com/class/the-midnight-quilt-show-season-5/',
+        'info_dict': {
+            'id': 'the-midnight-quilt-show-season-5',
+            'title': 'The Midnight Quilt Show Season 5',
+            'description': 'md5:113eda818e985d1a566625fb2f833b7a',
+        },
+        'playlist_count': 10,
+    }, {
+        'url': 'https://www.craftsy.com/class/sew-your-own-designer-handbag/',
+        'info_dict': {
+            'id': 'sew-your-own-designer-handbag',
+            'title': 'Sew Your Own Designer Handbag',
+            'description': 'md5:8270d0ef5427d3c895a27351aeaac276',
+        },
+        'playlist_mincount': 1,
+    }, {
+        'url': 'https://www.craftsy.com/class/all-access-estes-park-wool-market/',
+        'info_dict': {
+            'id': 'all-access-estes-park-wool-market',
+            'title': 'All Access: Estes Park Wool Market',
+            'description': 'md5:aded1bd8d38ae2fae4dae936c0ae01e7',
+        },
+        'playlist_count': 6,
+    }]
+
+    def _real_extract(self, url):
+        video_name = self._match_id(url)
+        webpage = self._download_webpage(url, video_name)
+
+        extra_data = get_element_by_id('vidstore-classes_class-video-player-js-extra', webpage)
+        video_data = json.loads(js_to_json(self._search_regex(
+            r'class_video_player_vars\s*=\s*({.*})\s*;',
+            extra_data, 'video data')))
+
+        account_id = traverse_obj(video_data, ('video_player', 'bc_account_id'))
+        class_title = video_data.get('class_title')
+        description = self._html_search_meta(['og:description', 'description'], webpage, default=None)
+
+        is_free = video_data.get('is_free')
+        user_has_access = video_data.get('user_has_access')
+        class_preview = traverse_obj(video_data, ('video_player', 'class_preview'))
```

Comment:
none of these need to be variables (since they are only used once). Just use them directly where needed

Reply:
class_preview is used in 3 places, so I keep it.

https://github.com/yt-dlp/yt-dlp/blob/17872d78c1ade7461d8429067cd00f56c289e325/yt_dlp/extractor/craftsy.py#L54-L60

Just changed the line to be just before IF.